java/http/java-http-consumer/src/main/java/HttpConsumer.java

```diff
@@ -121,19 +121,27 @@ public void subscribe() throws URISyntaxException, IOException, InterruptedExcep
     }
 
     public void run() throws InterruptedException {
+        Long messageCount = this.config.getMessageCount();
+        if ( messageCount == null ) {
+            messageCount = (Long.MAX_VALUE / this.config.getPollInterval() - 60_000L);
+        }
```

Comment:
What not just run forever when it is null?

Reply:
Above now removed